components/script/dom/globalscope.rs

```diff
@@ -263,6 +263,11 @@ impl GlobalScope {
         unreachable!();
     }
 
+    /// Get the origin URL for the current global object, shorthand for GlobalScope::current().get_url()
+    pub fn origin() -> ServoUrl {
```

Comment:
This function is a misnomer then -- it's not really getting the origin of an environment settings object, since it returns a `ServoUrl`.